lidar_frame: "velodyne"
lidar_topic: "/points_raw"

# output things ================ >
odom_lidar_topic: "/odom_lidar" # sensor_msgs::PointCloud2
output_odom_topic: "/auto_odom" # nav_msgs::Odometry

# will republish point with now timestamp
debug_print: true
use_odom: false
use_imu: false

# whether inverse the imu axis
imu_upside_down: false
odom_inverse: true
incremental_voxel_update: true

# setting -1 mean it will output the whole ndt map ===> FOR THE WHOLE MAP
# setting 10 mean 10*frame_point_num will be saved ===> FOR FAST LOC
save_frame_point: 10

baselink2LiDAR: [0, 0, 0, 0, 0, 0] # x,y,z, roll, pitch, yaw

# parameter's for ndt =========================== >
# Cell Size while mapping using ND (meters) (default 1.0)
resolution: 0.5

# Increment value between iterations while matching (default 0.1)
step_size: 0.05

# Value to decide convergence between iterations (meters) (default 0.01)
trans_epsilon: 0.01

# Maximum number of iterations before stopping matching (default 30)
max_iterations: 30

# Voxel Grid Size of the input scan (downsampling) (default 1.0)
leaf_size: 0.5

# Ignore points closer than this value (meters) (default 5.0)
min_scan_range: 0.5

# Ignore points far than this value (meters) (default 200.0)
max_scan_range: 20.0

# Minimum distance between points to be added to the final map (default 1.0)
min_add_scan_shift: 0.2

# parameter's for segmentation on ground pt =========================== >
process_ground_cloud: true
pub_ground_cloud: true

# when window < 3, elevation height initial dis
initial_distance: 0.1
# when elevation > max distance, threshold to max_distance, if it's stair this param may need some changes
max_distance: 2.5

# threshold for window 2kb+1
max_window_size: 8
# voxel size smaller is better but the
cell_size: 1
# based on the scenarios slope degree ==> TODO through height and window size as paper said
slope: 0.1